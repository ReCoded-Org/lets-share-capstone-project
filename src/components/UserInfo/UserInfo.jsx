import React from "react";
import Link from "next/link";
import Image from "next/image";
import { useTranslation } from "next-i18next";

// The props is necessary for the actual data... ***TO BE USED IN THE FUTURE***
const UserInfo = ({ userInfo }) => {
    //  ***to be removed later, FOR TEST PURPOSES ONLY***
    const dummyUser = {
        name: "John Doe",
        lang: "En",
        email: "blabla@example.com",
        location: "Kocaeli",
        phone: "01234567890",
        image: "/images/noPicInfo.svg",
    };

    return (
        <div className='m-3 flex flex-col items-center justify-center md:flex-row '>
            <div className='m-4 flex flex-row items-center justify-center md:flex-col '>
                <Image
                    className=' rounded-full p-2 '
                    height={"100%"}
                    width={"100%"}
                    src={dummyUser.image}
                />
                <span className=' m-1 flex justify-center p-1 text-center '>
                    {dummyUser.name}
                    <br />
                    {dummyUser.location}
                </span>
            </div>
            <div className='m-4 flex   flex-row justify-center rounded-lg bg-userInfoBg text-sm  md:h-28 md:w-[60%] '>
                <div className=' flex  flex-1 flex-col flex-wrap items-start justify-around p-3 md:flex-row md:items-center  '>
                    <span className=' flex flex-col '>
                        <span className='mb-1 md:mb-2'>
                            Name: {dummyUser.name}
                        </span>
                        <span className='mb-1 md:mb-2'>
                            Location: {dummyUser.location}
                        </span>
                    </span>
                    <span className=' flex flex-col'>
                        <span className='mb-1 md:mb-2'>
                            Phone: {dummyUser.phone}
                        </span>
                        <span className='mb-1 md:mb-2'>
                            Email: {dummyUser.email}
                        </span>
                    </span>
                    <span>Language: {dummyUser.lang}</span>
                </div>
                <div className='flex h-[35px] flex-col p-1 hover:rounded-lg hover:bg-[#e8e8e8] md:h-[100%] md:justify-center'>
                    <Link href={"/editUser"}>
                        <a>
                            <Image
                                src={"/images/editInfo.svg"}
                                width={30}
                                height={30}
                            />
                        </a>
                    </Link>
                </div>
            </div>
        </div>
    );
};
export default UserInfo;
