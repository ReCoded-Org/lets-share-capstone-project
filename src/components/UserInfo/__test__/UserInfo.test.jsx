"use strict";

import renderer from "react-test-renderer";
import * as nextRouter from "next/router";
import UserInfo from "../UserInfo";

it("renders correctly", () => {
    const tree = renderer.create(<UserInfo />).toJSON();
    expect(tree).toMatchSnapshot();
});
function tryRoute(packacge) {
    packacge.useRouter = jest.fn();
    packacge.useRouter.mockImplementation(() => ({ route: "/" }));
}
tryRoute(nextRouter);
