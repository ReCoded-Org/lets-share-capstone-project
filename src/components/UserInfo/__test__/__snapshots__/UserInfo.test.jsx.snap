// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=" m-3 flex flex-col items-center justify-center font-primary md:flex-row "
>
  <div
    className="m-4 flex flex-row items-center justify-center md:flex-col "
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27100%27/%3e"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        alt="user-image"
        className=" rounded-full p-2 "
        data-nimg="intrinsic"
        decoding="async"
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="user-image"
          className=" rounded-full p-2 "
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="/images/noPicInfo.svg"
          srcSet="/images/noPicInfo.svg 1x, /images/noPicInfo.svg 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
    <div
      className=" flex flex-col justify-center md:flex-row "
    >
      <button
        onClick={[Function]}
      >
        <svg
          className="text-md -translate-x-8 translate-y-8 opacity-60 hover:fill-primary hover:opacity-100 md:translate-x-8 md:-translate-y-8"
          fill="currentColor"
          height={30}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={30}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <circle
            cx="12"
            cy="12"
            r="3.2"
          />
          <path
            d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
          />
        </svg>
      </button>
    </div>
    <span
      className="flex justify-center text-center font-semibold "
    >
      John Doe
    </span>
  </div>
  <div
    className="m-4 flex flex-row justify-center rounded-lg bg-[#ffffff] text-sm md:h-32 md:w-[60%] "
  >
    <div
      className="flex flex-1 flex-col flex-wrap items-start justify-around gap-4 p-3 text-base  lg:flex-row lg:items-center  "
    >
      <span
        className=" flex flex-col gap-4"
      >
        <span
          className=""
        >
          common.name
          : 
          John Doe
        </span>
        <span
          className=""
        >
          common.location
          : 
          Kocaeli
        </span>
      </span>
      <span
        className=" flex flex-col gap-4"
      >
        <span
          className=""
        >
          common.phone
          : 
          +90234567890
        </span>
        <span
          className=""
        >
          common.email
          : 
          blabla@example.com
        </span>
      </span>
      <span
        className="rounded-full border-2 border-solid border-[black] bg-userInfoBg p-2 hover:bg-[#dbdbdb]"
      >
        <a
          href="/change-password"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          profile.changePassword
        </a>
      </span>
    </div>
    <div
      className="flex h-[45px] w-[45px] flex-col items-center justify-center rounded-r-lg bg-[#e6e6e6] p-1 hover:bg-[#dbdbdb] md:h-[100%] md:justify-center"
    >
      <a
        href="/edit-profile"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2730%27%20height=%2730%27/%3e"
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="edit-icon"
            data-nimg="intrinsic"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="edit-icon"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/images/editInfo.svg"
              srcSet="/images/editInfo.svg 1x, /images/editInfo.svg 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
      </a>
    </div>
  </div>
</div>
`;
